pysteps/nowcasts/linda.py

```diff
@@ -0,0 +1,1410 @@
+"""
+pysteps.nowcasts.linda
+======================
+
+This module implements the Lagrangian INtegro-Difference equation model with
+Autoregression (LINDA). The model combines extrapolation, S-PROG, STEPS, ANVIL,
+integro-difference equation (IDE) and cell tracking methods. It can produce
+both deterministic and probabilistic nowcasts. LINDA is specifically designed
+for nowcasting intense localized rainfall. For this purpose, it is expected to
+give better forecast skill than S-PROG or STEPS.
+
+The model consists of the following components:
+
+1. feature detection to identify rain cells
+2. advection-based extrapolation
+3. autoregressive integrated ARI(p,1) process for growth and decay of rainfall
+4. convolution to account for loss of predictability
+5. stochastic perturbations to simulate forecast errors
+
+LINDA utilizes a sparse feature-based representation of the input rain rate
+fields. This allows localization to cells containing intense rainfall. Building
+on extrapolation nowcast, the temporal evolution of rainfall is modeled in the
+Lagrangian coordinates. Using the ARI process is adapted from ANVIL
+:cite:`PCLH2020`, and the convolution is adapted from the integro-difference
+equation (IDE) models proposed in :cite:`FW2005` and :cite:`XWF2005`. The
+combination of these two approaches essentially replaces the cascade-based
+autoregressive process used in S-PROG and STEPS. Using the convolution gives
+several advantages such as the ability to handle anisotropic structure, domain
+boundaries and missing data. Based on the marginal distribution and covariance
+structure of forecast errors, localized perturbations are generated by adapting
+the short-space Fourier transform (SSFT) methodology developed in
+:cite:`NBSG2017`.
+
+.. autosummary::
+    :toctree: ../generated/
+
+    forecast
+"""
+
+import time
+import warnings
+
+try:
+    import dask
+
+    DASK_IMPORTED = True
+except ImportError:
+    DASK_IMPORTED = False
+import numpy as np
+from scipy.integrate import nquad
+from scipy.interpolate import interp1d
+from scipy import optimize as opt
+from scipy.signal import convolve
+from scipy import stats
+from pysteps import extrapolation, feature, noise
+
+warnings.filterwarnings("ignore")
+
+
+def forecast(
+    precip_fields,
+    advection_field,
+    timesteps,
+    feature_method="blob",
+    feature_kwargs=None,
+    ari_order=1,
+    kernel_type="anisotropic",
+    localization_window_radius=None,
+    errdist_window_radius=None,
+    acf_window_radius=None,
+    extrap_method="semilagrangian",
+    extrap_kwargs=None,
+    add_perturbations=True,
+    pert_thrs=(0.5, 1.0),
+    num_ens_members=40,
```

Comment:
Ok, one last comment ;-)
Can we reduce the default number of ensemble members to something much lower such as 10? 40 members is too many for default settings (see the profiling results above).